src/services/conversation.py

```diff
@@ -0,0 +1,74 @@
+import os
+from collections.abc import AsyncGenerator
+from typing import Protocol
+
+from agents.common.data import Message
+from agents.graph import IGraph, KymaGraph
+from agents.memory.redis_checkpointer import RedisSaver, initialize_async_pool
+from utils.logging import get_logger
+from utils.models import LLM, ModelFactory
+
+logger = get_logger(__name__)
+
+REDIS_URL = f"{os.getenv('REDIS_URL')}/0"
```

Comment:
Ideally, this should be part of a separate module `Config`.

Reply:
yes, I already have this in mind for the next PR. I will create a TODO for this too.